htlcswitch/failure.go

```diff
@@ -97,6 +149,24 @@ func (s *SphinxErrorEncrypter) IntermediateEncrypt(reason lnwire.OpaqueReason) l
 	return s.EncryptError(false, reason)
 }
 
+// Type returns the identifier for a sphinx error encrypter.
+func (s *SphinxErrorEncrypter) Type() EncrypterType {
+	return EncrypterTypeSphinx
+}
+
+// Encode serializes the error encrypter to the provided io.Writer.
+func (s *SphinxErrorEncrypter) Encode(w io.Writer) error {
+	return s.OnionErrorEncrypter.Encode(w)
```

Comment:
Won't this method be implicit, since it embeds the `OnionErrorEncrypter`?

Reply:
I believe so, though in a follow-up PR this method will instead write the `ogPacket` so we don't persist the shared-secret directly. This allows us to bridge the two by having the methods defined and only altering the internal behavior.